weather.go

```diff
@@ -1,5 +1,115 @@
 package weather
 
-func Get(location, token string) ([]byte, error) {
-	return nil, nil
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math"
+	"net/http"
+)
+
+type Weather struct {
+	Weather []struct {
+		ID          int    `json:"id"`
+		Main        string `json:"main"`
+		Description string `json:"description"`
+		Icon        string `json:"icon"`
+	} `json:"weather"`
+	Main struct {
+		Temp      float64 `json:"temp"`
+		FeelsLike float64 `json:"feels_like"`
+		TempMin   float64 `json:"temp_min"`
+		TempMax   float64 `json:"temp_max"`
+		Pressure  int     `json:"pressure"`
+		Humidity  int     `json:"humidity"`
+	} `json:"main"`
+}
+
+type CityUnknown struct {
+	Cod     string `json:"cod"`
+	Message string `json:"message"`
+}
+
+type CliResponse struct {
+	OneWord string  `json:"oneword"`
+	Celcius float64 `json:"celcius"`
+}
+
+type Response struct {
+	OneWord string  `json:"oneword"`
+	Celcius float64 `json:"celcius"`
+}
+
+func CliOutput(token, location string) (output string) {
+
+	var r Response
+
+	resp := CallUrl(token, location)
+	weatherString, err := Get(resp)
+	if err != nil {
+		log.Fatal(err)
+	}
+	json.Unmarshal(weatherString, &r)
+
+	output = string(fmt.Sprintf("weather: %s\ncelcius: %v\n", r.OneWord, r.Celcius))
+	return output
+}
+
+func CallUrl(token, location string) http.Response {
+
+	var cu CityUnknown
+	domain := "api.openweathermap.org"
```

Comment:
This would prevent us from calling a fake server in a unit test, wouldn't it? We need some way to be able to inject the API URL we're calling, rather than hard-wiring it like this.